File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/courseref/CategorieSectionAdminService.java
package ma.zs.zyn.service.facade.admin.courseref;

import java.util.List;
import ma.zs.zyn.bean.core.courseref.CategorieSection;
import ma.zs.zyn.dao.criteria.core.courseref.CategorieSectionCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface CategorieSectionAdminService extends  IService<CategorieSection,CategorieSectionCriteria>  {

    List<CategorieSection> findBySuperCategorieSectionId(Long id);
    int deleteBySuperCategorieSectionId(Long id);
    long countBySuperCategorieSectionCode(String code);




}

File: frontend/src/app/shared/service/admin/courseref/CategorieSectionAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {CategorieSectionDto} from 'src/app/shared/model/courseref/CategorieSection.model';
import {CategorieSectionCriteria} from 'src/app/shared/criteria/courseref/CategorieSectionCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class CategorieSectionAdminService extends AbstractService<CategorieSectionDto, CategorieSectionCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): CategorieSectionDto {
        return new CategorieSectionDto();
    }

    public constrcutCriteria(): CategorieSectionCriteria {
        return new CategorieSectionCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/categorieSection/';
    }
}
